src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -939,8 +951,7 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
                     }
                     else if (!string.IsNullOrEmpty(arg) && SpecialCharacters.IsDash(arg[0]))
                     {
-                        Match m = argPattern.Match(arg);
-                        if (m.Success)
+                        if (arg.Contains(":"))
```

Comment:
Instead of walking the string twice, move the call to IndexOf to before the `if` and test `offset` >= 0 instead.